### PR TITLE
python3 fixes (SOFTWARE-4476)

### PR DESCRIPTION
**Pull request by edquist**
also, notably, rewrite convert_checksum more compactly

### CODE REVIEW

File: src/xcache-consistency-check [hunk 1]
```diff
@@ -178,7 +178,7 @@ def get_xrdpfc_print_ouput(filename):
     lines = []
     try:
         out = subprocess.Popen(['xrdpfc_print', '-v', filename], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-    except Exception, e:
+    except Exception as e:
         if "No such file or directory" in str(e):
```
Comment (matyasselmeci) [inline src/xcache-consistency-check:182]: This should be a separate PR but we shouldn't parse exception strings...

File: src/xcache-consistency-check [hunk 2]
```diff
@@ -251,8 +251,8 @@ def get_list_of_baskets_in_branch(branch, byte_ranges, is_full_file):
 
 def recursive_branch(branch, byte_ranges, is_full_file):
     list_baskets = []
-    if len(branch.keys()) > 0:
-        for sub_branch in branch.itervalues():
+    if len(list(branch.keys())) > 0:
```
Comment (matyasselmeci) [inline src/xcache-consistency-check:254]:
```suggestion
    if branch:
```
(if branch is a dictionary).

**Review by matyasselmeci — APPROVED** [review]

File: src/xcache-consistency-check (hunk 2 above)
Comment (edquist) [reply src/xcache-consistency-check:254]: > (if branch is a dictionary)

that was my question too.  It seemed like `branch` was getting used as a dictionary and in that case it's way better to test that directly vs making a list of keys and taking the length.  But i got bored trying to track down exactly what `branch` is.

It apparently is tied to whatever `uproot.open` returns - which is some kind of python Mapping.  I don't know if it's guaranteed to behave like a dict when evaluated like a bool.  If you have a reason to believe it will work, i'm all for it, otherwise i don't want to be responsible for breaking it :)

Comment (matyasselmeci) [reply src/xcache-consistency-check:254]: I don't have a reason to believe it will work, though I would consider it a bug in `uproot` if it didn't :-P

Comment (edquist) [reply src/xcache-consistency-check]: Ok, well then, I'm committing your suggestion.

And hoping for the best.

**Comment by edquist** [comment]
@brianhlin please go ahead and merge if you are satisfied.

I do not feel like a legit enough maintainer to hit the big green button myself.
